src/gsi/read_goesglm.f90

```diff
@@ -323,30 +322,7 @@ subroutine read_goesglm(nread,ndata,nodata,infile,obstype,lunout,twindin,sis)
   call closbf(lunin)
 
 ! Write header record and data to output file for further processing
-  allocate(iloc(ndata))
-  icount=0.
-  do i=1,maxobs
-     if(isort(i) > 0)then
-       icount=icount+1
-       iloc(icount)=isort(i)
-     end if
-  end do
-  if(ndata /= icount)then
```

Comment:
This test was removed because it did nothing because the whole isort, iloc part of the code did nothing.  icntpnt, iout and ndata are incremented by 1 for each observation (after any qc of the data) and should have the same value (unless ndata > maxobs which is handled properly in the code). isort will be equal to 1,2,3,...,ndata. Because of this isort(i) is always > 0, icount will be equal to ndata at the end of the icount loop and iloc will just be equal to 1,2,3,...ndata.  So moving the data from cdata_all to cdata_out is just unnecessary data movement and icount will always be equal to ndata.

Checking to ensure that icount == ndata is unnecessary since they will always be the same for this routine.